modules/v2/main.tf

```diff
@@ -14,6 +14,30 @@
  * limitations under the License.
  */
 
+locals {
+  create_service_account        = var.service_account != null ? false : var.create_service_account
+  service_account               = var.service_account != null ? var.service_account : (var.create_service_account ? google_service_account.sa[0].email : null)
+  service_account_project_roles = local.create_service_account ? var.service_account_project_roles : []
+  service_account_id = local.create_service_account ? {
+    id    = google_service_account.sa[0].account_id,
+    email = google_service_account.sa[0].email
+  } : {}
+}
+
+resource "google_service_account" "sa" {
+  project      = var.project_id
+  count        = local.create_service_account ? 1 : 0
+  account_id   = "${var.service_name}-service-account"
```

Comment:
I would also recommend adding in region - maybe "${var.service_name}-"${var.region}"-sa? We may also want to truncate like https://github.com/terraform-google-modules/terraform-google-kubernetes-engine/blob/270a5c7380dd9fe9c41779143ddbfd5bf32c8fff/sa.tf#L26

Reply:
Done!